engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -443,6 +446,12 @@ private AnalysisJobBuilder create(JobType job, SourceColumnMapping sourceColumnM
                         convertToDate(metadata.getUpdatedDate()), datastore.getName(), getSourceColumnPaths(job),
                         getSourceColumnTypes(job), variables, getMetadataProperties(metadata));
                 analysisJobBuilder.setAnalysisJobMetadata(immutableAnalysisJobMetadata);
+            } else {
+                if (variables.size() > 0) {
+                    final MutableAnalysisJobMetadata mutableAnalysisJobMetadata = new MutableAnalysisJobMetadata();
```

Comment:
Here the metadata is being overwritten but I think actually rather it should only set/add variables to any existing metadata. Or else if there was already some metadata (about other things than variables) it will get wiped by this action.